addons/website_sale/models/website.py

```diff
@@ -378,10 +341,73 @@ def sale_get_order(self, force_create=False, update_pricelist=False):
 
         return sale_order_sudo
 
+    def _prepare_sale_order_values(self, partner_sudo):
+        self.ensure_one()
+        addr = partner_sudo.address_get(['delivery'])
+        if not request.website.is_public_user():
+            # FIXME VFE why not use last_website_so_id field ?
+            last_sale_order = self.env['sale.order'].sudo().search(
+                [('partner_id', '=', partner_sudo.id)],
+                limit=1,
+                order="date_order desc, id desc",
+            )
+            if last_sale_order and last_sale_order.partner_shipping_id.active:  # first = me
+                addr['delivery'] = last_sale_order.partner_shipping_id.id
+
+        affiliate_id = request.session.get('affiliate_id')
+        salesperson_user_sudo = self.env['res.users'].sudo().browse(affiliate_id).exists()
+        if not salesperson_user_sudo:
+            salesperson_user_sudo = self.salesperson_id or partner_sudo.parent_id.user_id or partner_sudo.user_id
+
+        pricelist_id = self._get_current_pricelist_id(partner_sudo)
+        fiscal_position_id = self._get_current_fiscal_position_id(partner_sudo)
+
+        values = {
+            'company_id': self.company_id.id,
+
+            'fiscal_position_id': fiscal_position_id,
+            'partner_id': partner_sudo.id,
+            'partner_invoice_id': partner_sudo.id,
+            'partner_shipping_id': addr['delivery'],
+
+            'pricelist_id': pricelist_id,
+            'payment_term_id': self.sale_get_payment_term(partner_sudo),
+
+            'team_id': self.salesteam_id.id or partner_sudo.parent_id.team_id.id or partner_sudo.team_id.id,
+            'user_id': salesperson_user_sudo.id,
+            'website_id': self.id,
+        }
+
+        return values
+
+    @api.model
+    def sale_get_payment_term(self, partner):
+        pt = self.env.ref('account.account_payment_term_immediate', False).sudo()
```

Comment:
will break if record doesn't exist (None would be returned by ref call), sudo should be moved below.